azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/VirtualMachineApiLiveTest.java

```diff
@@ -0,0 +1,269 @@
+      VirtualMachine vm = api().create(getName(), id, getName(), LOCATION, getProperties(blob, nicName));
+      assertTrue(!vm.name().isEmpty());
+      String status = "Creating";
+      int stopCounter = 0;
+      while (status.equals("Creating")) {
+         try {
+            Thread.sleep(120 * 1000); // Creation takes long time. Recheck later
+         } catch (InterruptedException e) {
+            e.printStackTrace();
+         }
+         vm = api().get(getName());
+         status = vm.properties().provisioningState();
+         stopCounter++;
+         if (stopCounter == 15)
+            break;
+      }
+      status = vm.properties().provisioningState();
+
+      // Cannot be creating anymore. Should be succeeded or running but not failed.
+      assertTrue(!status.equals("Creating"));
+      assertTrue(!status.equals("Failed"));
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testGet() {
+      VirtualMachine vm = api().get(getName());
+      assertTrue(!vm.name().isEmpty());
+   }
+
+   @Test(dependsOnMethods = "testCreate")
+   public void testGetInstanceView() {
+      VirtualMachineInstance vmi = api().getInstanceDetails(getName());
+      assertTrue(!vmi.statuses().isEmpty());
+   }
+
+   @Test(dependsOnMethods = "testStart")
+   public void testStop() {
+      String status = "";
+      api().stop(getName());
+      int stopCounter = 0;
+      while (!status.equals("VM stopped")) {
+         List<VirtualMachineInstance.VirtualMachineStatus> statuses = api().getInstanceDetails(getName()).statuses();
+         for (int c = 0; c < statuses.size(); c++) {
+            if (statuses.get(c).code().substring(0, 10).equals("PowerState")) {
+               status = statuses.get(c).displayStatus();
+            }
+            try {
+               Thread.sleep(5 * 1000); // Stopping takes long time. Recheck later
+            } catch (InterruptedException e) {
+               e.printStackTrace();
+            }
+            stopCounter++;
+            if (stopCounter == 40)
+               break;
+         }
+      }
+      assertEquals(status, "VM stopped");
+
+   }
+
+   @Test(dependsOnMethods = "testGet")
+   public void testStart() {
+      api().start(getName());
+      String status = "";
+      int stopCounter = 0;
+      while (!status.equals("VM running")) {
+         List<VirtualMachineInstance.VirtualMachineStatus> statuses = api().getInstanceDetails(getName()).statuses();
+         for (int c = 0; c < statuses.size(); c++) {
+            if (statuses.get(c).code().substring(0, 10).equals("PowerState")) {
+               status = statuses.get(c).displayStatus();
+            }
+            try {
+               Thread.sleep(5 * 1000); // Starting takes long time. Recheck later
+            } catch (InterruptedException e) {
+               e.printStackTrace();
+            }
+            stopCounter++;
+            if (stopCounter == 40)
+               break;
+         }
+      }
```

Comment:
Also, instead of using a counter, let's better use a timeout and take benefit of the `Predicates2.retry`.